utils/spdx/src/main/kotlin/SpdxExpression.kt

```diff
@@ -571,8 +569,9 @@ class SpdxLicenseIdExpression(
     override fun validate(strictness: Strictness) {
         val isValid = SpdxConstants.isNotPresent(id) || when (strictness) {
             Strictness.ALLOW_ANY -> true
-            Strictness.ALLOW_DEPRECATED -> spdxLicense != null
-            Strictness.ALLOW_CURRENT, Strictness.ALLOW_LICENSEREF_EXCEPTIONS -> spdxLicense?.deprecated == false
+            Strictness.ALLOW_DEPRECATED -> SpdxLicense.forId(id) != null
+            Strictness.ALLOW_CURRENT, Strictness.ALLOW_LICENSEREF_EXCEPTIONS ->
+                SpdxLicense.forId(id)?.deprecated == false
```

Comment:
A lazy property has the benefit that it is only executed once if `validate` is called multiple times.

Reply:
It's probably not very likely that `validate()` is called multiple times for the same instance, but still done.